src/Games/NexusMods.Games.AdvancedInstaller.UI/Content/Left/ModContent/ModContentViewModel.cs

```diff
@@ -1,18 +1,132 @@
+
+            child.Item.Status = ModContentTreeEntryStatus.SelectingViaParent;
+            SelectChildrenRecursive(child);
+        }
+    }
+
+    public void DeselectChildrenRecursive(ModContentNode node)
+    {
+        foreach (var child in node.Children)
+        {
+            if (child.Item.Status != ModContentTreeEntryStatus.SelectingViaParent) continue;
+
+            child.Item.Status = ModContentTreeEntryStatus.Default;
+            DeselectChildrenRecursive(child);
+        }
+    }
+
+    #endregion utility
+
+    #region private
+
+    private readonly ReadOnlyObservableCollection<ModContentNode> _modContentTreeRoots;
+
+    /// <summary>
+    /// Populates the tree cache from a FileTreeNode structure representing the archive contents.
+    /// </summary>
+    /// <param name="cache">The tree entries cache, where new entries will be added.</param>
+    /// <param name="node">The root node of the filetree containing the archive contents that need to be added.</param>
+    private void PopulateModContentEntriesCache(
+        ISourceCache<IModContentTreeEntryViewModel, RelativePath> cache,
+        FileTreeNode<RelativePath, ModSourceFileEntry> node)
+    {
+        var allNodes = node.GetAllNodes();
+
+        // Populate the cache
+        cache.Edit(updater =>
+        {
+            // Create the root node
+            var root = new ModContentTreeEntryViewModel(RelativePath.Empty, true);
+            updater.AddOrUpdate(root);
+
+            foreach (var (_, fileTreeNode) in allNodes)
+            {
+                var relativePath = fileTreeNode.Path;
+                // All the tree leaf nodes are files, so it's a directory if it has children
+                var entry = new ModContentTreeEntryViewModel(relativePath, fileTreeNode.Children.Count > 0);
+                updater.AddOrUpdate(entry);
+            }
+        });
+    }
 
-    public ModContentViewModel(FileTreeNode<RelativePath, ModSourceFileEntry> archiveFiles, IAdvancedInstallerCoordinator coordinator)
+    /// <summary>
+    /// Required to display the tree in the TreeDataGrid.
+    /// </summary>
+    /// <param name="root">The root node of the view</param>
+    /// <returns></returns>
+    private static TreeDataGridSource CreateTreeDataGridSource(ModContentNode root)
     {
-        _vm = TreeEntryViewModel<ModSourceFileEntry>.FromFileTree(archiveFiles, coordinator);
+        return new TreeDataGridSource(root)
+        {
+            Columns =
+            {
+                new HierarchicalExpanderColumn<ModContentNode>(
+                    new TemplateColumn<ModContentNode>(null,
+                        new FuncDataTemplate<ModContentNode>((node, _) =>
+                            new ModContentTreeEntryView
+                            {
+                                // node can apparently be null even if it isn't nullable, likely for virtualization
+                                DataContext = node?.Item,
```

Comment:
You should add `// ReSharper disable once ConditionalAccessQualifierIsNonNullableAccordingToAPIContract` if that's the case.